router/internal/retrytransport/retry_transport.go

```diff
@@ -99,6 +100,8 @@ func (rt *RetryHTTPTransport) RoundTrip(req *http.Request) (*http.Response, erro
 		// drain the previous response before retrying
 		rt.drainBody(resp, requestLogger)
 
+		httpclient.IncrementRetryCount(req.Context())
```

Comment:
Is this called anywhere else? A user could have retry disabled.

Reply:
Its only called if retry is enabled.